code/__DEFINES/is_helpers.dm

```diff
@@ -11,6 +11,9 @@
 
 #define isweakref(D) (istype(D, /datum/weakref))
 
+// simple check whether or not a player is a guest using their key and computer_id to build the key and then make a string comparison with the actual key
+#define IS_GUEST_KEY(key, computer_id)	("[key]" == "Guest-[computer_id]")
```

Comment:
After pondering it, to stay on the safe side I think you should check if it starts with `Guest-` and then if the remainder of it is a number.

Reply:
Or just only check that it starts with `Guest-`

Reply:
Yeah just check if it starts with `Guest-`